core/src/main/java/com/cloud/info/ConsoleProxyInfo.java

```diff
@@ -36,33 +36,33 @@ public ConsoleProxyInfo(int proxyUrlPort) {
     public ConsoleProxyInfo(boolean sslEnabled, String proxyIpAddress, int port, int proxyUrlPort, String consoleProxyUrlDomain) {
         this.sslEnabled = sslEnabled;
 
-        if (sslEnabled) {
-            StringBuffer sb = new StringBuffer();
-            if (consoleProxyUrlDomain.startsWith("*")) {
-                sb.append(proxyIpAddress);
-                for (int i = 0; i < proxyIpAddress.length(); i++)
-                    if (sb.charAt(i) == '.')
-                        sb.setCharAt(i, '-');
-                sb.append(consoleProxyUrlDomain.substring(1));//skip the *
-            } else {
-                //LB address
-                sb.append(consoleProxyUrlDomain);
-            }
-            proxyAddress = sb.toString();
-            proxyPort = port;
-            this.proxyUrlPort = proxyUrlPort;
+        StringBuffer sb = new StringBuffer();
```

Comment:
Would you mind extracting this block that creates the "proxyAddress" to a method? Then, we could have unit tests and documentation for it.